sql/parse/parse.go

```diff
@@ -1786,7 +1786,7 @@ func convertVal(v *sqlparser.SQLVal) (sql.Expression, error) {
 		}
 		return expression.NewLiteral(val, sql.LongBlob), nil
 	case sqlparser.ValArg:
-		return expression.NewLiteral(string(v.Val), sql.LongText), nil
+		return expression.NewBindVar(strings.TrimPrefix(string(v.Val), ":")), nil
```

Comment:
Do named bindvars need to be distinct? If so, need an analyzer func to check that

Reply:
No, they're also support for a different syntax, `SELECT * FROM test WHERE id = :id`, and the identifier `id` could be used multiple times in the query. IDs are auto-assigned by the tokenizer for the `?` syntax though, and those are distinct.

There's a problem though. The parser will parse a query that uses both `?` and `:name` style variables and with the current approach it will run into a conflict if someone names one of their vars `:v1`, for example. Longer-term we should distinguish between the two types of variables in the parser I think.